### ISSUE TITLE
Race condition with configstore

### ISSUE THREAD

**Issue by blakeembrey** [yeoman/update-notifier#68]
From https://github.com/typings/typings/issues/80#issuecomment-166671308.

I'm not the original author, see linked issue, but it appears some sort of race condition might be occurring when checking for updates in multiple processes that are being spawned. If this is the case, a solution might be to set the config store before running the check, then having subsequent stores use the latest config to check if a different check script is already currently executing. This would avoid having multiple updates executing at once, though I can't tell if it'd solve the core issue.

**Comment by sindresorhus** [comment]
I don't have the time right now to look into this, but it sounds like an issue with [`write-file-atomic`](https://github.com/npm/write-file-atomic). It should be able to handle concurrent operations. That's the whole point of that module. Probably worth opening an issue over there.

@timjroberts What Node.js version are you on?

**Comment by timjroberts** [comment]
@sindresorhus We're using Node 4.0.0

**Comment by SimenB** [comment]
@sindresorhus this is probably the same issue I had in sindresorhus/del-cli#7. Should I still create a new issue?

**Comment by sindresorhus** [comment]
@SimenB Are you using it from multiple processes?

**Comment by SimenB** [comment]
Updated my comment, but yes, we're using `npm-run-all` in parallel

**Comment by sindresorhus** [comment]
Alright. It seems pretty clear now this is a [`write-file-atomic`](https://github.com/npm/write-file-atomic) race issue. Could you open an issue there?

**Comment by sindresorhus** [comment]
Fixed by https://github.com/npm/write-file-atomic/pull/13.
